File: Arrays/06-Binary Search In Array/Binary-Search.cpp
#include <iostream>
using namespace std;

int binarySearch(int arr[], int n, int x)
{

    int start = 0;
    int end = n - 1;

    while (start <= end)
    {

        int mid = (start + end) / 2;

        if (arr[mid] == x)
        {
            return mid;
        }
        else if (x > arr[mid])
        {
            start = mid + 1;
        }
        else
        {
            end = mid - 1;
        }
    }

    return -1;
}

int main()
{

    int arr[] = {3, 5, 6, 8, 44, 66, 70, 80, 100};

    int n = sizeof(arr) / sizeof(arr[0]);
    int x = 5;

    int ans = binarySearch(arr, n, x);

    cout << ans << endl;

    return 0;
}

File: Arrays/06-Binary Search In Array/Binary-Search.js
function binarySearch(arr, x) {
  let start = 0;
  let end = arr.length - 1;

  while (start <= end) {
    let mid = Math.floor((start + end) / 2);

    if (arr[mid] == x) {
      return mid;
    } else if (x > arr[mid]) {
      start = mid + 1;
    } else {
      end = mid - 1;
    }
  }

  return -1;
}

let arr = [3, 5, 6, 8, 44, 66, 70, 80, 100];

let ans = binarySearch(arr, 70);

console.log(ans);
